Shared.Tests/CloudinaryTest.cs

```diff
@@ -1191,81 +1243,85 @@ public void TestGetResource()
         public void TestGetResourceWithMetadata()
         {
             // should allow get resource metadata
+            var publicId = $"{m_apiTest}_TestGetResourceWithMetadata";
 
             ImageUploadParams uploadParams = new ImageUploadParams()
             {
                 File = new FileDescription(m_testImagePath),
-                EagerTransforms = new List<Transformation>() { new Transformation().Crop("scale").Width(2.0) },
-                PublicId = "testgetresource2"
+                EagerTransforms = new List<Transformation>() { m_simpleTransformation },
+                PublicId = publicId,
+                Tags = m_apiTag
             };
 
             m_cloudinary.Upload(uploadParams);
 
             GetResourceResult getResult = m_cloudinary.GetResource(
-                new GetResourceParams("testgetresource2")
+                new GetResourceParams(publicId)
                 {
                     Metadata = true
                 });
 
             Assert.IsNotNull(getResult);
-            Assert.AreEqual("testgetresource2", getResult.PublicId);
+            Assert.AreEqual(publicId, getResult.PublicId);
             Assert.NotNull(getResult.Metadata);
         }
 
         [Test]
         public void TestDeleteDerived()
         {
             // should allow deleting derived resource
+            var publicId = $"{m_apiTest}_TestDeleteDerived";
 
             ImageUploadParams uploadParams = new ImageUploadParams()
             {
                 File = new FileDescription(m_testImagePath),
-                EagerTransforms = new List<Transformation>() { new Transformation().Width(101).Crop("scale") },
-                PublicId = "testdeletederived"
+                EagerTransforms = new List<Transformation>() { m_simpleTransformation },
+                PublicId = publicId,
+                Tags = m_apiTag
             };
 
             m_cloudinary.Upload(uploadParams);
 
-            GetResourceResult resource = m_cloudinary.GetResource("testdeletederived");
+            GetResourceResult resource = m_cloudinary.GetResource(publicId);
 
             Assert.IsNotNull(resource);
             Assert.AreEqual(1, resource.Derived.Length);
 
-            DelDerivedResResult delDerivedResult =
-                m_cloudinary.DeleteDerivedResources(resource.Derived[0].Id);
-
+            DelDerivedResResult delDerivedResult = m_cloudinary.DeleteDerivedResources(resource.Derived[0].Id);
             Assert.AreEqual(1, delDerivedResult.Deleted.Values.Count);
 
-            resource = m_cloudinary.GetResource("testdeletederived");
-
+            resource = m_cloudinary.GetResource(publicId);
             Assert.IsFalse(String.IsNullOrEmpty(resource.PublicId));
         }
 
         [Test]
         public void TestDelete()
         {
             // should allow deleting resources
+            var publicId = $"{m_apiTest}_TestDelete";
+            var rndString = $"{m_apiTest}_SomeRandomString";
```

Comment:
make it random, maybe add some helper function to generate a random string easily